scripts/surface_charge/_surface_charge_calculator.py

```diff
@@ -0,0 +1,144 @@
+#
+# This script can be used for any purpose without limitation subject to the
+# conditions at http://www.ccdc.cam.ac.uk/Community/Pages/Licences/v2.aspx
+#
+# This permission notice and the following statement of attribution must be
+# included in all copies or substantial portions of this script.
+#
+# The following line states a licence feature that is required to show this script in Mercury and Hermes script menus.
+# Created 18/08/2024 by Alex Moldovan
+# ccdc-licence-features not-applicable
+
+import os
+import warnings
+from typing import List, Tuple
+
+import numpy as np
+from ccdc.io import CrystalReader
+from ccdc.particle import Surface
+from ccdc.utilities import HTMLReport
+
+
+class SurfaceCharge:
+    def __init__(self, crystal, use_existing_charges: bool = False):
+        if use_existing_charges:
+            # if crystal.molecule.assign_partial_charges() is False:
+            #     warnings.warn(f"Gasteiger charges could not be assigned to molecule: {crystal.identifier}",
+            #                   RuntimeWarning)
+            raise NotImplementedError("Use existing charges instead. Current implementation only supports Gasteiger.")
```

Comment:
does this prevent user from using charges read from .mol2 file?

Reply:
At the moment yes. We need to investigate the user supplied charges as I know some of ours tool (such as VH) can override these so I was going to implement a separate feature for handling user supplied charges and checking they are valid.